### PR TITLE
Feature/skip xml [9139]

### PR DESCRIPTION
**Pull request by imontesino**
Added environment variable SKIP_DEFAULT_XML_FILE to skip over searching for the default file. Also added a unit test for it.

### CODE REVIEW

File: src/cpp/rtps/xmlparser/XMLProfileManager.cpp [hunk 1]
```diff
@@ -171,15 +171,28 @@ void XMLProfileManager::loadDefaultXMLFile()
...
     if (const char* file_path = std::getenv(DEFAULT_FASTRTPS_ENV_VARIABLE))
     {
         loadXMLFile(file_path);
     }
-#endif // ifdef _WIN32
 
-    // Try to load the default XML file.
-    loadXMLFile(DEFAULT_FASTRTPS_PROFILES);
+    // Try to load the default XML file unless set to false
+    if (!std::getenv("SKIP_DEFAULT_XML_FILE"))
```
Comment (EduPonz) [inline src/cpp/rtps/xmlparser/XMLProfileManager.cpp:191]: On `WIN32` you are checking whether the value of the variable is `1`, however there is no check in Linux. I'd add the same for consistency

**Review by EduPonz — CHANGES_REQUESTED** [review]
Partial review

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/2495/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/2309/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/2502/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/2497/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/2311/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/2504/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/2500/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/2314/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/2507/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/2657/)
* Linux aarch64[Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux%20aarch64/69/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/2470/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/2667/)

**Comment by MiguelCompany** [comment]
New tests failing on two platforms. I canceled the jobs for the other two

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/2672/)
* Linux aarch64[Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux%20aarch64/75/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/2476/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/2682/)

**Comment by MiguelCompany** [comment]
Seems the unit tests have been fixed. We need a PR against docs adding the new environment variable before merging this one.

**Comment by imontesino** [comment]
Addded documentation in eProsima/Fast-RTPS-docs#195

**Pull request merged**